compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -270,6 +271,31 @@ impl<'a> Parser<'a> {
         };
         self.last_unexpected_token_span = Some(self.token.span);
         let mut err = self.struct_span_err(self.token.span, &msg_exp);
+
+        // Add suggestion for a missing closing angle bracket if '>' is included in expected_tokens
+        // and there are unclosed angle brackets
+        if !self.angle_args_span_stack.is_empty()
+            && expected.iter().any(|tok| match tok {
```

Comment:
I think we can reduce some false positives if we also check whether `inedible` contains `=`, because that's the only site where the suggestion actually works, right?